### PR TITLE
SDIT-996 Non-associations-api now uses upsert request for sync

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/SqsIntegrationTestBase.kt [hunk 1]
```diff
@@ -100,9 +100,11 @@ class SqsIntegrationTestBase : TestBase() {
   internal val awsSqsSentencingOffenderEventsClient by lazy { sentencingOffenderEventsQueue.sqsClient }
   internal val awsSqsSentencingOffenderEventsDlqClient by lazy { sentencingOffenderEventsQueue.sqsDlqClient }
 
-  internal val nonAssociationsOffenderEventsClient by lazy { nonAssociationsOffenderEventsQueue.sqsClient }
   internal val nonAssociationsOffenderEventsQueue by lazy { hmppsQueueService.findByQueueId("eventnonassociations") as HmppsQueue }
   internal val nonAssociationsQueueOffenderEventsUrl by lazy { nonAssociationsOffenderEventsQueue.queueUrl }
+  internal val nonAssociationsQueueOffenderEventsDlqUrl by lazy { nonAssociationsOffenderEventsQueue.dlqUrl as String }
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/integration/SqsIntegrationTestBase.kt:105]: Renamed to match other queue client names

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/NomisApiService.kt [hunk 2]
```diff
@@ -366,28 +367,16 @@ constructor(
...
-    )
-}
+fun NonAssociationResponse.toUpsertSyncRequest() =
+  UpsertSyncRequest(
+    firstPrisonerNumber = offenderNo,
+    firstPrisonerReason = UpsertSyncRequest.FirstPrisonerReason.valueOf(reason),
+    secondPrisonerNumber = nsOffenderNo,
+    secondPrisonerReason = UpsertSyncRequest.SecondPrisonerReason.valueOf(recipReason!!),
+    restrictionType = UpsertSyncRequest.RestrictionType.valueOf(type),
+    active = true, // TODO Fix
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/NomisApiService.kt:377]: Active will be removed shortly - but need non-associations-api to be updated first

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt [hunk 3]
```diff
@@ -303,6 +337,12 @@ class NonAssociationsSynchronisationIntTest : SqsIntegrationTestBase() {
   }
 }
 
+fun SqsAsyncClient.waitForMessageCountOnQueue(queueUrl: String, messageCount: Int) =
+  await untilCallTo {
+    this.countAllMessagesOnQueue(queueUrl)
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt:342]: don't need "this" - will remove in next PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt [hunk 4]
```diff
@@ -185,11 +189,11 @@ class NonAssociationsSynchronisationIntTest : SqsIntegrationTestBase() {
...
         fun `will not create the non-association in the non-associations service`() {
-          nonAssociationsApi.verify(exactly(0), postRequestedFor(urlPathEqualTo("/sync")))
+          nonAssociationsApi.verify(exactly(0), putRequestedFor(urlPathEqualTo("/sync")))
         }
 
         @Test
-        fun `will not create telemetry tracking`() {
+        fun `will not create telemetry tracking success`() {
           verify(telemetryClient, Times(0)).trackEvent(
             eq("non-association-created-synchronisation-success"),
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt:198]: the customevent name has changed to non-association-upserted-synchronisation-success

**Review by steverendell — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt (hunk 4 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt:198]: ha ha yes - I thought I'd done a full find and replace - will fix in next PR

**Pull request merged**